android/expoview/src/main/java/versioned/host/exp/exponent/modules/api/ScreenOrientationModule.java

```diff
@@ -55,47 +61,220 @@ public void onCatalystInstanceDestroy() {
       return;
     }
-
-    activity.setRequestedOrientation(convertToOrientationEnum(orientation));
     promise.resolve(null);
   }
 
   @ReactMethod
-  public void doesSupportAsync(String orientation, Promise promise) {
+  public void lockPlatformAsync(int orientationAttr, Promise promise) {
+    Activity activity = getCurrentActivity();
+
     try {
-      convertToOrientationEnum(orientation);
-    } catch (JSApplicationIllegalArgumentException exception) {
-      promise.reject(exception);
+      activity.setRequestedOrientation(orientationAttr);
+    } catch (Exception e) {
+      promise.reject(ERR_SCREEN_ORIENTATION, "Could not apply the ScreenOrientation platform lock: " + orientationAttr, e);
       return;
     }
-    promise.resolve(true);
+    promise.resolve(null);
+
+  }
+
+  @ReactMethod
+  public void unlockAsync(Promise promise) {
+    lockAsync(OrientationLock.DEFAULT.toString(), promise);
+  }
+
+  @ReactMethod
+  public void getOrientationAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+
+    try {
+      Orientation orientation = getScreenOrientation(activity);
+      promise.resolve(orientation.toString());
+    } catch (Exception e) {
+      promise.reject(ERR_SCREEN_ORIENTATION, "Could not get the current screen orientation", e);
+    }
+  }
+
+  @ReactMethod
+  public void getOrientationLockAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+
+    try {
+      int orientationAttr = activity.getRequestedOrientation();
+      OrientationLock orientationLock = orientationLockNativeToJS(orientationAttr);
+      promise.resolve(orientationLock.toString());
+    } catch (Exception e) {
+      promise.reject(ERR_SCREEN_ORIENTATION, "Could not get the current screen orientation lock", e);
+    }
+
+  }
+
+  @ReactMethod
+  public void getPlatformOrientationLockAsync(Promise promise) {
+    Activity activity = getCurrentActivity();
+    try {
+      promise.resolve(activity.getRequestedOrientation());
+    } catch (Exception e) {
+      promise.reject(ERR_SCREEN_ORIENTATION, "Could not get the current screen orientation platform lock", e);
+    }
+
+  }
+
+  @ReactMethod
+  public void supportsOrientationLockAsync(String orientationLockStr, Promise promise){
+    try {
+      // If we can get the native orientation value from the given string without throwing, we resolve with true
+      OrientationLock lockJS = OrientationLock.valueOf(orientationLockStr);
+      orientationLockJSToNative(lockJS);
+      promise.resolve(true);
+    } catch (Exception e) {
+      promise.resolve(false);
+    }
+  }
+
+  // https://stackoverflow.com/questions/10380989/how-do-i-get-the-current-orientation-activityinfo-screen-orientation-of-an-a
+  // Will not work in all cases as surface rotation is not standardized across android devices, but this is best effort
+  private Orientation getScreenOrientation(Activity activity) {
```

Comment:
this seems hacky but I'm ok with including it since it's tested in test-suite. I do think we should be clear in the docs about the difference between this & getting the window dimensions to compare/figure out aspect ratio (which you can do with a different module)